File: tia-client/src/main/java/com/teamscale/tia.client/AgentHttpRequestFailedException.java
package com.teamscale.tia.client;

/**
 * Thrown if communicating with the agent via HTTP fails. The underlying reason can be either a network problem or an
 * internal error in the agent. Users of this library should report these exceptions appropriately so the underlying
 * problems can be addressed.
 */
public class AgentHttpRequestFailedException extends Exception {

	public AgentHttpRequestFailedException(String message) {
		super(message);
	}

	public AgentHttpRequestFailedException(String message, Throwable cause) {
		super(message, cause);
	}

}

File: tia-client/src/main/kotlin/com/teamscale/tia/client/AgentHttpRequestFailedException.kt
package com.teamscale.tia.client

/**
 * Thrown if communicating with the agent via HTTP fails. The underlying reason can be either a network problem or an
 * internal error in the agent. Users of this library should report these exceptions appropriately so the underlying
 * problems can be addressed.
 */
class AgentHttpRequestFailedException : Exception {
	constructor(message: String?) : super(message)

	constructor(message: String?, cause: Throwable?) : super(message, cause)
}
